src/pal/src/map/map.cpp

```diff
@@ -2241,7 +2241,9 @@ MAPmmapAndRecord(
     PAL_ERROR palError = NO_ERROR;
     LPVOID pvBaseAddress = NULL;
 
-    pvBaseAddress = mmap(addr, len, prot, flags, fd, offset);
+    off_t adjust = offset & (VIRTUAL_PAGE_MASK);
+
+    pvBaseAddress = mmap(static_cast<char *>(addr) - adjust, len + adjust, prot, flags, fd, offset - adjust);
```

Comment:
You can use BIT64 && FEATURE_PAL to enable it for ARM64 and AMD64, since the cross tools are built for the same bitness of host and target architecture. I guess that if we changed that in the future, we would introduce something like TARGET_BIT64.